core/templates/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -17,81 +17,101 @@
  * any learner answer info.
  */
 
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/statistics/LearnerAnswerInfoObjectFactory.ts');
-require('domain/statistics/LearnerAnswerDetailsObjectFactory.ts');
+import {ExplorationDataService} from
+  './exploration-data.service';
+import {UrlInterpolationService} from
+  '../../../domain/utilities/url-interpolation.service';
+import {LearnerAnswerInfo} from
+  '../../../domain/statistics/LearnerAnswerInfoObjectFactory';
+import {LearnerAnswerDetails} from
+  '../../../domain/statistics/LearnerAnswerDetailsObjectFactory';
+import {downgradeInjectable} from
+  '@angular/upgrade/static';  
+import {Injectable} from
+  '@angular/core';
+import {HttpClient} from
+  '@angular/common/http';
 
-angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
-  '$http', '$q', 'ExplorationDataService', 'LearnerAnswerDetailsObjectFactory',
-  'LearnerAnswerInfoObjectFactory', 'UrlInterpolationService',
-  function(
-      $http, $q, ExplorationDataService, LearnerAnswerDetailsObjectFactory,
-      LearnerAnswerInfoObjectFactory, UrlInterpolationService) {
-    var _expId = ExplorationDataService.explorationId;
-    var _data = [];
-    var learnerAnswerInfoData = null;
-    var LEARNER_ANSWER_INFO_DATA_URL = (
+@Injectable( {
+  providedIn: 'root'
+})
+export class LearnerAnswerDetailsDataService {
+  
+  constructor(
+    private explorationDataService:ExplorationDataService,
+    private urlInterpolationService: UrlInterpolationService,
+    private http:HttpClient,
+  ){
+    
+  }
+    _expId;
+    _data;
+    learnerAnswerInfoData=null;
+    LEARNER_ANSWER_INFO_DATA_URL=(
       '/learneranswerinfohandler/learner_answer_details/<entity_type>/' +
-      '<entity_id>');
-
-    var _fetchLearnerAnswerInfoData = function() {
-      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
-        LEARNER_ANSWER_INFO_DATA_URL, {
-          entity_type: 'exploration',
-          entity_id: _expId});
-
-      return $http.get(learnerAnswerInfoDataUrl);
-    };
+      '<entity_id>'
+    );
 
-    var _deleteLearnerAnswerInfo = function(
-        entityId, stateName, learnerAnswerInfoId) {
-      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
-        LEARNER_ANSWER_INFO_DATA_URL, {
-          entity_type: 'exploration',
-          entity_id: entityId});
-      return $http['delete'](learnerAnswerInfoDataUrl, {
+   private _fetchLearnerAnswerInfoData=()=>{
+      var learnerAnswerInfoDataUrl=this.urlInterpolationService.interpolateUrl(
+        this.LEARNER_ANSWER_INFO_DATA_URL, {
+        entity_type : 'exploration',
+        entity_id : this._expId
+        });
+        return this.http.get(learnerAnswerInfoDataUrl);
```

Comment:
We convert the `http.get/post` etc to pormises.
```suggestion
        return this.http.get(learnerAnswerInfoDataUrl).toPromise();
```

Reply:
Done